src/utils/sample/uniform.rs

```diff
@@ -112,7 +112,11 @@ fn sample_bits_uniform(nr_bits: usize) -> Vec<u8> {
     } else {
         byte_vector = vec![0u8; nr_bits / 8 + 1];
     }
-    rng.fill_bytes(&mut byte_vector);
+    let mut res = rng.try_fill_bytes(&mut byte_vector);
+    while res.is_err() {
+        let first_unfilled_byte = find_first_unfilled_byte(&byte_vector);
+        res = rng.try_fill_bytes(&mut byte_vector[first_unfilled_byte..]);
```

Comment:
Maybe add a comment: COntinue at first zero byte, this may potentially discard previously sampled values, but still stays uniform in the distribution